src/misc/msccl/msccl_lifecycle.cc

```diff
@@ -451,6 +451,28 @@ static ncclResult_t mscclFallBackSavedParams() {
   return ncclSuccess;
 }
 
+#ifdef ENABLE_MSCCLPP
+inline bool isMscclppSupportedType(ncclDataType_t dataType) {
+  switch (dataType) {
+  case ncclInt8:
+  case ncclUint8:
+  case ncclFloat16:
+  case ncclInt32:
+  case ncclUint32:
+  case ncclFloat32:
+  case ncclInt64:
+  case ncclUint64:
+  case ncclFloat64:
+#ifdef RCCL_BFLOAT16
+  case ncclBfloat16:
+#endif
```

Comment:
Only these are supported, see https://github.com/microsoft/mscclpp/blob/main/apps/nccl/src/nccl.cu#L497
```suggestion
  case ncclFloat16:
  case ncclInt32:
  case ncclUint32:
  case ncclFloat32:
#ifdef RCCL_BFLOAT16
  case ncclBfloat16:
#endif
```

Reply:
@nusislam mscclpp doesn't have the same restriction on data type for `ncclAllGather`. Should I reflect that here, or can I keep this global restriction?

Reply:
Update docs with these missing types.